src/renderer/processor.rs

```diff
@@ -801,7 +803,16 @@ impl<'a> Processor<'a> {
                 let template = self.tera.get_template(tpl_name)?;
                 self.macros.add_macros_from_template(&self.tera, template)?;
                 self.call_stack.push_include_frame(tpl_name, template);
-                let result = self.render_body(&template.ast)?;
+                let result = {
+                    let mut processor = Processor::new(
+                        template,
+                        self.tera,
+                        self.context,
+                        self.should_escape,
+                        &mut self.call_stack
```

Comment:
Not sure if it's the issue but this is not wanted. It would mean that an included template can modify the parent context.

Reply:
Can templates modify the context? I thought they only created variables in stack frames, as the context is a non-mutable reference.